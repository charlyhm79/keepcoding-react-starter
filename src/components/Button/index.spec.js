import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Button, { getHalf } from './';

describe('Button test suite', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('getHalf', () => {
    expect(getHalf(4)).toBe(2);
    expect(getHalf(1)).toBe(0.5);
    expect(getHalf(-1)).toBe(-0.5);
    expect(getHalf(10)).toBe(5);
    expect(getHalf(1000)).toBe(500);
  });

  it('Matchs snapshot', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
