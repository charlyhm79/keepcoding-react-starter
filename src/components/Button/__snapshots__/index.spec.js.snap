// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test suite Matchs snapshot 1`] = `
<button
  className="sc-bdVaJa gAwOrD"
  to={undefined}
/>
`;
